packages/editor-ui/src/views/NodeView.vue

```diff
@@ -2261,25 +2201,103 @@ export default mixins(
 						setStateDirty: true,
 					});
 					if (!this.suspendRecordingDetachedConnections) {
-						this.historyStore.pushCommandToUndo(new AddConnectionCommand(connectionData, this));
+						this.historyStore.pushCommandToUndo(new AddConnectionCommand(connectionData));
+					}
+					if (!this.isReadOnly) {
+						NodeViewUtils.addConnectionActionsOverlay(
+							info.connection,
+							() => {
+								this.activeConnection = null;
+								this.__deleteJSPlumbConnection(info.connection);
+							},
+							() => {
+								insertNodeAfterSelected({
+									sourceId: info.sourceEndpoint.parameters.nodeId,
+									index: sourceInfo.index,
+									connection: info.connection,
+									eventSource: 'node_connection_action',
+								});
+							},
+						);
+					}
+				} catch (e) {
+					console.error(e); // eslint-disable-line no-console
+				}
+			});
+
+			this.instance.bind(EVENT_DRAG_MOVE, () => {
+				this.instance?.connections.forEach((connection) => {
+					NodeViewUtils.showOrHideItemsLabel(connection);
+					NodeViewUtils.showOrHideMidpointArrow(connection);
+
+					Object.values(connection.overlays).forEach((overlay) => {
+						if (!overlay.canvas) return;
+						this.instance?.repaint(overlay.canvas);
+					});
+				});
+			});
+			this.instance.bind(EVENT_CONNECTION_MOUSEOVER, (connection: Connection) => {
+				try {
+					if (this.exitTimer !== undefined) {
```

Comment:
the timers worked better when they were not global, but tied to connection state.. for example if you put two connections next to each other and hover quickly between them, it's easy to get stuck in a hover state without actions.. not the biggest issue..